### ISSUE TITLE
[Quote] Gray 90/100 theme link wrong color

### ISSUE THREAD

**Issue by RichKummer** [carbon-design-system/carbon-for-ibm-dotcom#8705]
### Description

For React version, in Gray 90 and Gray 100 themes, the link color for the CTA is wrong. The color should match Web components. Currently the React version shows `$interactive01` as the color token, when it should be using `$link-01`.

**React:**
<img width="986" alt="Screen Shot 2022-04-12 at 5 35 02 PM" src="https://user-images.githubusercontent.com/15643582/163058503-4c81fc39-2ba4-473c-925b-df11983fbdfd.png">
<img width="640" alt="Screen Shot 2022-04-12 at 5 35 21 PM" src="https://user-images.githubusercontent.com/15643582/163058520-c58b8d15-0e40-4a70-9ab7-357a36854ae1.png">

**Web Components:**
<img width="1258" alt="Screen Shot 2022-04-12 at 5 30 59 PM" src="https://user-images.githubusercontent.com/15643582/163058357-b30e843a-93d6-40f5-95ed-e4d479977345.png">

<img width="571" alt="Screen Shot 2022-04-12 at 5 36 29 PM" src="https://user-images.githubusercontent.com/15643582/163058598-6cd34802-676b-49dd-93cb-6abaf21afe04.png">

### Component(s) impacted

Quote

### Browser

Chrome, Safari

### Carbon for IBM.com version

v1.33-rc.0

### Severity

Severity 4 = The problem is not visible to or noticeable to an average user. Affects minor functionality, no workaround needed.

### Application/website

C4IBM.com

### Package

@carbon/ibmdotcom-react

### CodeSandbox example

https://codesandbox.io/s/github/carbon-design-system/carbon-for-ibm-dotcom/tree/main/packages/web-components/examples/codesandbox/components/quote

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by oliviaflory** [comment]
@RichKummer double checking specs

**Comment by RichKummer** [comment]
Updated reference to `$link-01` per the specs
